src/support/execAsync.ts

```diff
@@ -0,0 +1,69 @@
+// https://github.com/microsoft/vscode-docker/blob/main/src/utils/execAsync.ts
+import * as stream from 'stream';
+import { AccumulatorStream, Shell, StreamSpawnOptions } from '@microsoft/vscode-container-client';
+import { spawnStreamAsync } from './spawnStreamAsync';
+import * as cp from 'child_process';
+import { CancellationToken } from 'vscode';
+
+type Progress = (content: string, err: boolean) => void;
+
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+export type ExecError = Error & { code: any, signal: any, stdErrHandled: boolean };
+
+export type ExecAsyncOutput = { stdout: string, stderr: string };
+
+export async function execAsync(command: string, options?: cp.ExecOptions & { stdin?: string, cancellationToken?: CancellationToken }, progress?: Progress): Promise<ExecAsyncOutput> {
```

Comment:
could you remind which lines we had to change here to accommodate the error handling? Basically why we had to copy this file over?

Reply:
We found it necessary to include the `execAsync.ts` file by copying it into our project. Unfortunately, this file was not included in the newly imported dependency package that manages most of the command line execution. Note that no modifications were made to the `execAsync.ts` file itself.

To enhance our error-handling capabilities, we duplicated the `spawnStreamAsync.ts` file and integrated it into our extension. The original version of this file lacked detailed error messages; it returned vague "ChildProcessError" messages, which didn't allow for precise error management. To rectify this, we decided to import a version of the file and made specific changes. Specifically, we modified lines [284 and 286](https://github.com/microsoft/vscode-docker-extensibility/blob/main/packages/vscode-container-client/src/utils/spawnStreamAsync.ts#L288) into code `resolve();` to provide accurate details about the errors that occurred during execution.